src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -180,37 +180,32 @@ def _try_handle_high_volume_task_trigger(ts: datetime):
     reference_datetime = ts.replace(second=0, microsecond=0)
     reference_ts = int(reference_datetime.timestamp())
 
-    last_ts = redis_client.get(HIGH_VOLUME_LAST_TRIGGER_TS_KEY)
+    # Since GETSET is atomic this acts as a guard against another consumer
+    # picking up the minute rollover
+    last_ts = redis_client.getset(HIGH_VOLUME_LAST_TRIGGER_TS_KEY, reference_ts)
     if last_ts is not None:
         last_ts = int(last_ts)
 
     # Do nothing until the message we process moves across the minute boundary
     if last_ts == reference_ts:
         return
 
-    try:
-        lock = locks.get("sentry.monitors.task_trigger", duration=5)
-        with lock.acquire():
-            # Track the delay from the true time, ideally this should be pretty
-            # close, but in the case of a backlog, this will be much higher
-            total_delay = reference_ts - datetime.now().timestamp()
-
-            metrics.incr("monitors.task.triggered_via_high_volume_clock")
-            metrics.gauge("monitors.task.high_volume_clock_delay", total_delay)
-
-            # If more than exactly a minute has passed then we've skipped a
-            # task run, report that to sentry, it is a problem.
-            if last_ts is not None and last_ts + 60 != reference_ts:
-                with sentry_sdk.push_scope() as scope:
-                    scope.set_extra("last_ts", last_ts)
-                    scope.set_extra("reference_ts", reference_ts)
-                    sentry_sdk.capture_message("Monitor task dispatch minute skipped")
-
-            _dispatch_tasks(ts)
-            redis_client.set(HIGH_VOLUME_LAST_TRIGGER_TS_KEY, reference_ts)
-    except UnableToAcquireLock:
-        # Another message processor is handling this. Nothing to do
-        pass
+    # Track the delay from the true time, ideally this should be pretty
+    # close, but in the case of a backlog, this will be much higher
+    total_delay = reference_ts - datetime.now().timestamp()
+
+    metrics.incr("monitors.task.triggered_via_high_volume_clock")
+    metrics.gauge("monitors.task.high_volume_clock_delay", total_delay)
+
+    # If more than exactly a minute has passed then we've skipped a
+    # task run, report that to sentry, it is a problem.
+    if last_ts is not None and last_ts + 60 != reference_ts:
```

Comment:
if `last_ts` is `None` do we want to process? or wait for a full minute for it to clear out/increment?
feel like we should process in the case of deployments etc

Reply:
I think it's OK to process it yeah.

Reply:
We do already, this is just guarding the exception logic for when it skips

Reply:
oh right i was reading this backwards